### PR TITLE
Add simple transmission spectrum demo

### PR DESCRIPTION
**Pull request by bmorris3**
This PR: 
* adds an `Opacity` class for finding the available opacities on disk, loading them into memory, and generating a jitted interpolator for the opacity grid
* adds a fast function for creating a small synthetic opacity file
* adds narrative docs for how to work with `Opacity` objects, including:
  * loading from disk
  * inspecting and slicing the xarray object within
  * constructing a jitted and vectorized opacity interpolator
  * cropping the opacity grid on a wavelength range
* adds a function for Heng & Kitzmann (2017)'s isothermal transmission spectrum
* brings together the points above for a demo in the docs producing a transmission spectrum

The docs add in this PR include: 
* an intro to Opacity: https://shone--4.org.readthedocs.build/en/4/shone/opacities.html
* transmission demo: https://shone--4.org.readthedocs.build/en/4/shone/examples/simple_transmission.html

### CODE REVIEW

File: shone/opacity/dace.py [hunk 1]
```diff
@@ -169,7 +169,10 @@ def bin_files(members):
 
 
 def get_opacity_dir_path_molecule(isotopologue, linelist):
-    return glob(os.path.join('tmp', isotopologue + '__' + linelist + "*e2b"))[0]
+    find_e2b = glob(os.path.join('tmp', isotopologue + '__' + linelist + "*e2b"))
+    if len(find_e2b):
+        return find_e2b[0]
+    return glob(os.path.join('tmp', isotopologue + '__' + linelist + "*"))[0]
```
Comment (Hoeijmakers) [inline shone/opacity/dace.py:175]: This line 175 as well as the old line 179 do not catch the case that for some reason the linelist file/tmp folder do not exist, and instead give a non-descript IndexError.

Comment (Hoeijmakers) [reply shone/opacity/dace.py:175]: In the download-molecule example, I actually encounter an error on line 221: Unique_wavelengths doesn't seem to be defined.

File: docs/shone/examples/opacities.rst [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+
+    opacity_sample = opacity.grid.sel(
+        dict(
+            pressure=10,  # [bar]
+            temperature=200  # [K]
+        )
+    )
+
+    plt.semilogy(
+        opacity_sample.wavelength, opacity_sample.opacity,
```
Comment (Hoeijmakers) [inline docs/shone/examples/opacities.rst:136]: `opacity_sample.opacity` is wrong. It should be `opacity_sample`, equal to what is the source code of the plot that comes after this cell.

File: docs/shone/examples/opacities.rst [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+
+    # get a jitted 3D interpolator over wavelength, temperature, pressure:
+    interp_opacity = opacity.get_interpolator()
+
+Now you can get the opacity at wavelengths, temperatures, and pressures that weren't on
+the grid:
+
+.. code-block:: python
+
+    wavelength = np.linspace(1, 5, 500)  # [µm]
```
Comment (Hoeijmakers) [inline docs/shone/examples/opacities.rst:184]: Line 184 requires an `import numpy as np` somewhere.

File: docs/shone/examples/opacities.rst [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+    pressure = 0.3  # [bar]
+    temperature = 555  # [K]
+
+    example_opacity = interp_opacity(wavelength, temperature, pressure)
+
+    plt.semilogy(wavelength, example_opacity, label=f"T={temperature} K")
```
Comment (Hoeijmakers) [inline docs/shone/examples/opacities.rst:190]: I suggest to overplot the opacity functions at the temperatures of the grid points to compare. If the synthetic opacity is T-dependent ofc.

**Review by Hoeijmakers — CHANGES_REQUESTED** [review]
See my comments in-line. Some tiny issues!

File: docs/shone/examples/opacities.rst (hunk 4 above)
Comment (bmorris3) [reply docs/shone/examples/opacities.rst:190]: Wouldn't you say that's sort of redundant? The point of H&K 2017 is that R ~ ln(kappa).

Comment (Hoeijmakers) [reply docs/shone/examples/opacities.rst:193]: No this was mostly just for visual purposes. To show that the interpolated opacity lands between two grid entries. Not necessary though.

File: docs/shone/examples/simple_transmission.rst [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+temperatures. The function ``interp_opacity`` does not take vector
+arguments for temperature or pressure, but we can easily vectorize it
+with `~jax.vmap` like this:
+
+.. code-block:: python
+
+    # interpolate for a range of wavelengths at one pressure and temperature:
+    temperature = np.array([200, 400, 600, 800])
+
+    example_opacity = vmap(
```
Comment (Hoeijmakers) [inline docs/shone/examples/simple_transmission.rst:144]: This call to vmap doesn't work in the example if `from jax import numpy as jnp, vmap` has not been repeated.

Comment (Hoeijmakers) [reply docs/shone/examples/simple_transmission.rst:144]: I added it!

**Review by Hoeijmakers — APPROVED** [review]
I fixed a missing import statement and approve!

**Comment by bmorris3** [comment]
Thanks!

**Pull request merged**